nixos/modules/services/web-apps/mastodon.nix

```
@@ -575,6 +583,38 @@ in {
       path = with pkgs; [ file imagemagick ffmpeg ];
     };
 
+
+  systemd.services.mastodon-media-prune =
+    let
+      cfg = config.services.mastodon;
+    in
+    lib.mkIf cfg.mediaAutoRemove {
+      description = "Mastodon media prune";
+      environment = lib.filterAttrs (n: _: n != "PATH") config.systemd.services.mastodon-web.environment;
+      serviceConfig = {
+        Type = "oneshot";
+        # Remove remote media attachments and preview cards older than one month.
+        ExecStart = ''
+          ${cfg.package}/bin/tootctl media remove --days=30
+          ${cfg.package}/bin/tootctl preview_cards remove --days=30
+        '';
+        User = cfg.user;
+        Group = cfg.group;
+        EnvironmentFile = "/var/lib/mastodon/.secrets_env";
+        PrivateTmp = true;
+      };
+    };
+
+  systemd.timers.mastodon-media-prune = lib.mkIf cfg.mediaAutoRemove {
+    description = "Mastodon media prune";
+    wantedBy = [ "timers.target" ];
+    timerConfig = {
+      OnCalendar = "*-*-* 00:00:00"; # every day
```

```suggestion
      OnCalendar = "daily";
```

See `systemd.time(7)`